packages/vsx-registry/src/browser/vsx-extensions-widget.ts

```diff
@@ -57,27 +59,60 @@ export class VSXExtensionsWidget extends SourceTreeWidget {
         this.addClass('theia-vsx-extensions');
 
         this.id = generateExtensionWidgetId(this.options.id);
-        const title = this.options.title ?? this.computeTitle();
-        this.title.label = title;
-        this.title.caption = title;
 
         this.toDispose.push(this.extensionsSource);
         this.source = this.extensionsSource;
+
+        this.computeTitle();
+
+        this.toDispose.push(this.source.onDidChange(() => {
+            this.computeTitle();
+        }));
+    }
+
+    private _badgeCount: number | undefined = undefined;
+    get badge(): number | undefined {
+        return this._badgeCount;
+    }
+
+    set badge(count: number | undefined) {
+        this._badgeCount = count;
+        this.onDidChangeBadge.fire();
     }
 
-    protected computeTitle(): string {
+    public onDidChangeBadge: Emitter<void> = new Emitter<void>();
+
+    protected async computeTitle(): Promise<void> {
+        let label: string;
         switch (this.options.id) {
             case VSXExtensionsSourceOptions.INSTALLED:
-                return nls.localizeByDefault('Installed');
+                label = nls.localizeByDefault('Installed');
+                break;
             case VSXExtensionsSourceOptions.BUILT_IN:
-                return nls.localizeByDefault('Built-in');
+                label = nls.localizeByDefault('Built-in');
+                break;
             case VSXExtensionsSourceOptions.RECOMMENDED:
-                return nls.localizeByDefault('Recommended');
+                label = nls.localizeByDefault('Recommended');
+                break;
             case VSXExtensionsSourceOptions.SEARCH_RESULT:
-                return nls.localize('theia/vsx-registry/openVSX', 'Open VSX Registry');
+                label = nls.localize('theia/vsx-registry/openVSX', 'Open VSX Registry');
+                break;
             default:
-                return '';
+                label = '';
+        }
+        const title = `${label}`;
+        this.title.label = title;
+        this.title.caption = title;
+
+        this.badge = await this.resolveCount();
+    }
+
+    protected async resolveCount(): Promise<number | undefined> {
+        if (this.options.id !== VSXExtensionsSourceOptions.SEARCH_RESULT) {
+            const elements = await this.source?.getElements() || [];
+            return [...elements].length;
         }
+        return undefined;
```

Comment:
The changes do not properly handle a `0` count:

![image](https://user-images.githubusercontent.com/40359487/173394117-1bf3404f-13c7-4938-bca1-7e71e7fc4715.png)

In this case I would still expect `0` to be displayed for recommended and installed.

Reply:
I have submitted a new commit to address these comments !!
Thanks @vince-fugnitto !!